setup.py

```diff
@@ -337,7 +335,7 @@ def run(self):
     if os.environ.get("PANDAS_CI", "0") == "1":
         extra_compile_args.append("-Werror")
     if debugging_symbols_requested:
-        extra_compile_args.append("-g")
+        extra_compile_args.append("-g3")
```

Comment:
A bit tangential but g3 provides visibility into macros with gcc, which is useful for this PR